src/operations/scripts.rs

```diff
@@ -0,0 +1,200 @@
+        return Err(new_error(&format!(
+            "{:?} does not exist. You may want to call `dmenv develop` now",
+            src_path
+        )));
+    }
+    #[cfg(windows)]
+    {
+        match mode {
+            Safe => safe_copy(&src_path, &dest_path),
+            Override => copy(&src_path, &dest_path),
+        }
+    }
+    #[cfg(unix)]
+    {
+        symlink(&src_path, &dest_path, mode)
+    }
+}
+
+#[cfg(windows)]
+fn safe_copy(src_path: &PathBuf, dest_path: &PathBuf) -> Result<(), Error> {
+    if dest_path.exists() {
+        return Err(new_error(&format!("{:?} already exists", src_path)));
+    }
+    copy(src_path, dest_path)
+}
+
+#[cfg(windows)]
+fn copy(src_path: &PathBuf, dest_path: &PathBuf) -> Result<(), Error> {
+    std::fs::copy(src_path, dest_path).map_err(|e| {
+        new_error(&format!(
+            "Could not copy from {:?} to {:?}: {}",
+            src_path, dest_path, e
+        ))
+    })?;
+    Ok(())
+}
+
+#[cfg(unix)]
+fn symlink(
+    src_path: &PathBuf,
+    dest_path: &PathBuf,
+    mode: &ProcessScriptsMode,
+) -> Result<(), Error> {
+    match mode {
+        // Note: we assume it is "safe" to change the target of an existing
+        // symlink
+        Safe => delete_if_link(dest_path),
+        // Note: we need to delete dest_path for unix::fs::symlink to work
+        // later on
+        Override => delete_if_exists(dest_path),
+    }?;
+    println!(
+        "{} -> {}",
+        dest_path.to_string_lossy(),
+        src_path.to_string_lossy()
+    );
+    std::os::unix::fs::symlink(src_path, dest_path).map_err(|e| {
+        new_error(&format!(
+            "Could not create link from {:?} to {:?}: {}",
+            dest_path, src_path, e
+        ))
+    })
+}
+
+#[cfg(unix)]
+fn delete_if_link(path: &PathBuf) -> Result<(), Error> {
+    // This will make an error if the path does not exist,
+    // but also if we don't have permission to read the path.
+    // In both cases, we want to return early - in the later,
+    // the code will probably fail when creating the symlink
+    // later on.
+    let meta = std::fs::symlink_metadata(path);
+    if meta.is_err() {
+        return Ok(());
+    };
+    let meta = meta.unwrap();
+    if !meta.file_type().is_symlink() {
+        return Err(new_error(&format!(
+            "{:?} exists and is *not* a symlink",
+            path
+        )));
+    };
+    std::fs::remove_file(path).map_err(|e| {
+        new_error(&format!(
+            "Could not remove existing symlink {:?}: {}",
+            path, e
+        ))
+    })?;
+    Ok(())
+}
+
+#[cfg(unix)]
+fn delete_if_exists(path: &PathBuf) -> Result<(), Error> {
+    if path.exists() {
+        std::fs::remove_file(path)
+            .map_err(|e| new_error(&format!("Could not remove existing file {:?}: {}", path, e)))?;
+    }
+    Ok(())
+}
+
+fn find_egg_info(project_path: &PathBuf) -> Result<PathBuf, Error> {
+    let pattern = format!("{}/*.egg-info", project_path.to_string_lossy());
```

Comment:
Hhhhmm this is sadly a bug :/
Say I'm a user in a locale with a funny Windows codepage, and I like to put characters that can't be translated to valid UTF-8 in my project paths: your glob will fail (or worse, silently do the wrong thing if I'm a particularly twisted user and also have a valid project folder with a unicode replacement character in it!).

Arguably this is a bug in `glob` for not providing some kind of `glob_at` option that takes a base directory, but in this case it's probably okay for you to change the current working directory (altho I'm sure you know that can cause its own set of problems...)?

Reply:
Your're right. Let's not use `glob` at all, then

Reply:
:+1: